app/services/solr_mapper.rb

```diff
@@ -88,6 +104,39 @@ def funders_ids_field
     metadata['funding_references']&.pluck('funder_identifier')&.compact || []
   end
 
+  def descriptions_by_type_field(description_types)
+    metadata['descriptions']&.filter_map do |d|
+      d['description'].truncate(32_000) if description_types.include?(d['description_type'])
+    end || []
+  end
+
+  def rights_uris_field
+    metadata['rights_list'].pluck('rights_uri')&.uniq&.compact if metadata['rights_list'].present?
+  end
+
+  # Names of organizations affiliated with both contributors and creators of a dataset
+  def affilation_names_field
+    affiliation_names_for_role('creators').concat(affiliation_names_for_role('contributors')).uniq
+  end
+
+  # Extract the year from the date for temporal coverage
+  # If the date is a range, store a sequence of years from beginning to end
+  def temporal_field
+    metadata['dates']&.filter_map do |date|
+      parse_date(date['date']) if date['date_type'] == 'Coverage'
+    end&.flatten
+  end
+
+  # Return an array of dates based on the parsing of the date string
+  def parse_date(date_value)
```

Comment:
Instead of continuing to expand this class, distinct concerns like dates should be split into separate mapper classes. (Considering this in a separate PR is fine.)

Reply:
I will set up a new issue for that, agreed.